src/main/java/org/ecocean/identity/IBEISIA.java

```diff
@@ -1510,6 +1511,12 @@ public static JSONObject processCallback(String taskID, JSONObject resp, String
                		if (taskParameters == null) taskParameters = new JSONObject();
                		JSONObject tp = new JSONObject();
                     JSONObject mf = new JSONObject();
+            		Encounter enc=myShepherd.getEncounter(encUUID);
+                    if(enc!=null && enc.getLocationID()!=null) {
+                    	ArrayList<String> locationIDs = new ArrayList<String>(); 
+                    	List<String> matchTheseLocationIDs = LocationID.getIDForParentAndChildren(enc.getLocationID(), locationIDs,null); 
+                    	mf.put("locationIds",matchTheseLocationIDs);
+                    }
```

Comment:
this is one possible fix. i think we can also consider this (or variation):
```
if ((parentTask != null) && (parentTask.getParameters() != null)) 
   mf = parentTask.getParameters().optJSONObject("matchingSetFilter");
```
which, i believe, will get the matchingSetFilter which comes from how this task was initiated. in my testing, this does seem to bring in the encounter-specific location when **submitting a new** encounter. but should double-check this.

Reply:
Thank you! Please push this change to this branch.